glusterd2/commands/volumes/volume-expand-txn.go

```diff
@@ -238,3 +240,147 @@ func updateVolinfoOnExpand(c transaction.TxnCtx) error {
 
 	return nil
 }
+
+// getDeviceAvailableSize gets the device size and vgName using device Path
+func getDeviceAvailableSize(devicePath string) (string, uint64) {
+	devices, err := deviceutils.GetDevices(gdctx.MyUUID.String())
+	deviceVgName := strings.Split(devicePath, "/")[2]
+	if err != nil {
+		return "", 0
```

Comment:
This error might be due to etcd connection breakdown, and returning 0 instead of the error might lead to false information.